src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -196,7 +196,13 @@ void DlgPrefWaveform::slotSetWaveformType(int index) {
     if (index < 0) {
         return;
     }
+    // This regenerates the waveforms twice because of a bug found on Windows
+    // where the first one fails.
+    // The problem is that the window of the widget thinks that it is not exposed.
+    // (https://doc.qt.io/qt-5/qwindow.html#exposeEvent )
+    // TODO: Remove this when it has been fixed upstream.
     WaveformWidgetFactory::instance()->setWidgetTypeFromHandle(index);
+    WaveformWidgetFactory::instance()->setWidgetTypeFromHandle(index, true);
```

Comment:
I think we  should guard it with the QT version and the effected OS. 
This way we will find this spot easily when we consider the next QT update.

Reply:
No. GL needs to be started at least once.
As I said, I also was reproducing the problem if starting with nonGl and then switching to a GL one from the preferences.
Ok, I can add if QT >= 5.14. Is that ok?

Reply:
Yes, thank you.